Sources/StreamChatUI/ChatMessageList/ChatThreadVC.swift

```diff
@@ -378,6 +384,25 @@ open class _ChatThreadVC<ExtraData: ExtraDataTypes>:
     }
 
     // MARK: - _ChatMessageControllerDelegate
+    
+    public func messageController(
+        _ controller: _ChatMessageController<ExtraData>,
+        didChangeMessage change: EntityChange<_ChatMessage<ExtraData>>
+    ) {
+        let indexPath = IndexPath(row: messageController.replies.count, section: 0)
+        
+        let listChange: ListChange<_ChatMessage<ExtraData>>
+        switch change {
+        case .create(let item):
+            listChange = .insert(item, index: indexPath)
+        case .update(let item):
+            listChange = .update(item, index: indexPath)
+        case .remove(let item):
+            listChange = .remove(item, index: indexPath)
+        }
+        
+        updateMessages(with: [listChange])
```

Comment:
we were supposed to call `reloadData` but `didChangeMessage` is triggered when thread root message is updated which can happen multiple times. Reloading single cell sounds better (the index calculations are straightforward since thread root is always at the end of message list)